test/inductor/test_torchinductor_dynamic_shapes.py

```diff
@@ -219,6 +219,17 @@ def f(x, b):
         opt_r = opt_f(x, b)
         self.assertEqual(r, opt_r)
 
+    @torch._dynamo.config.patch(
+        capture_dynamic_output_shape_ops=True
+    )
+    def test_nonzero_no_realloc(self, device):
+        @torch.compile(fullgraph=True, dynamic=True)
+        def f(x, y):
+            z = x.nonzero()
+            return torch.split(z, [y.size(0)])
+
+        f(torch.tensor([1, 0, 1, 1, 0, 1, 0]), torch.randn(4))
```

Comment:
hmm, I tried running this test locally (without your PR) to see what the error is and it isn't erroring for me. Was this supposed to fail?

Reply:
No, it doesn't fail, unfortunately. You have to look at TORCH_LOGS=dynamic to see if the freshly allocated unbacked SymInt gets unified away or no